src/App.js

```diff
@@ -0,0 +1,111 @@
+import DocumentList from "./component/DocumentList.js";
+import Document from "./component/Document.js";
+import { initRouter, push } from "./core/router.js";
+import nestedObjectToArray from "./core/nestedObjectToArray.js";
+import { getItem, setItem } from "./core/storage.js";
+import { request } from "./core/api.js";
+import editor from "./component/Editor.js";
+
+export default function App({ $target }) {
+  this.state = getItem("tempSaveKey") || alert("새로고침을 해주세요");
+
+  this.setState = async (nextState) => {
+    this.state = nextState;
+    await documentList.setState(this.state);
+    await document.setState(this.state);
+  };
+
+  this.fetchDocument = async () => {
+    if (getItem("tempSaveKey") == null) {
+      const serverDocument = nestedObjectToArray(
+        await request("/documents", {
+          method: "GET",
+        })
+      );
+      setItem("tempSaveKey", {
+        documentList: serverDocument,
+        currentDocumentId: serverDocument[0].id,
+      });
```

Comment:
P5;

물론 해당 코드 또한 잘 작성되었지만, `request`가 하는 일이 워낙 크기 때문에 변수로 한번 짚고 코드를 작성해도 좋았을 것 같아요.

```suggestion
      const serverDocumentList = await request('/document', {...});
      const currentDocumentList = nestedObjectToArray(serverDocumentList);
      
      setItem("tempSaveKey", {
        documentList: currentDocumentList,
        currentDocumentId: currentDocumentList[0].id,
      });
```